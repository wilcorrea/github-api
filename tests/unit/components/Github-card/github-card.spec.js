import { shallowMount } from '@vue/test-utils'
// noinspection NpmUsedModulesInstalled
import GithubCard from 'src/components/Github-card/Github-card.vue'

describe('Test Github Card', function () {
  describe('invalid user', function () {
    const wrapper = shallowMount(GithubCard)

    // validate available
    it('check available computed', () => {
      expect(wrapper.vm.available).toBe(false)
    })

    // create and compare snapshots
    it('should render content correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('invalid user', function () {
    const user = {
      login: 'wilcorrea',
      avatar_url: undefined,
      html_url: undefined,
      company: undefined,
      location: undefined,
      public_repos: undefined,
      repos_url: undefined,
      followers: undefined,
      following_url: undefined,
      following: undefined
    }
    const wrapper = shallowMount(GithubCard, { attrs: { user } })

    // validate available
    it('check available computed', () => {
      expect(wrapper.vm.available).toBe(true)
    })

    // match html rendered
    it('renders the correct markup', () => {
      expect(wrapper.html()).toContain(user.login)
    })

    // create and compare snapshots
    it('should render content correctly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

})
