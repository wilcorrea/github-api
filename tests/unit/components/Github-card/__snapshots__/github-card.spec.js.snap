// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Github Card invalid user should render content correctly 1`] = `undefined`;

exports[`Test Github Card invalid user should render content correctly 2`] = `
"<div class=\\"github-card\\"><div class=\\"github-avatar\\"><img alt=\\"\\"></div> <div class=\\"github-information\\"><span class=\\"name\\"><a>
        wilcorrea
      </a></span> <span class=\\"company\\">
      
    </span> <span>
      
    </span> <hr> <span>
      
      <a>Repositories</a></span> <span>
      
      <a>Followers</a></span> <span>
      
      <a>
        Following
      </a></span></div></div>"
`;
